### DISCUSSION TITLE
conda: `github-linguist` tries to use ruby in rubygems directory

### DISCUSSION THREAD

**Discussion by sdruskat** [github-linguist/linguist#6914] [Q&A] [answered]
I managed to install `github-linguist` in conda (mamba 1.5.5, conda 23.11.0) on Ubuntu (WSL2), using this method:

```bash
mamba create -n ghl python=3.11
mamba activate ghl
mamba install compilers ruby libgit2=1.7.1 cmake pkg-config icu zlib
gem install github-linguist -- --use-system-libraries
```

🎉 

Now, when running `github-linguist .` in a local git repo, I get an error:

**`<conda-envs>/ghl/share/rubygems/bin/github-linguist: 6: exec: <conda-envs>/ghl/share/rubygems/bin/ruby: not found`**.

I'm wondering why `github-linguist` is looking for a ruby binary in `rubygems/`?

FWIW: I can easily fix this doing `ln -s <conda-envs>/ghl/bin/ruby <conda-envs>/ghl/share/rubygems/bin/ruby`, but would like to understand why the error occurs in the first place...

Is this due to the way I installed ruby? Can I call ghl providing the ruby version to use? 🤷 
Thanks for any pointers!

**Comment by lildude** [comment, marked as answer]
I have no idea what mamba is or how it works, but this sounds like something related to the gem environment which isn't setting the correct variables or some-such.

I use [mise](https://mise.jdx.dev/) on macOS and my `github-linguist` command lands in `$HOME/.local/share/gem/bin/github-linguist` with this being the first line:

```
➜ head -1 /Users/lildude/.local/share/gem/bin/github-linguist
#!/Users/lildude/.local/share/mise/installs/ruby/3.3.3/bin/ruby
➜ 
```

This comes from the `RUBY EXECUTABLE` configuration in my `gem environment`:

```console
➜ gem environment
RubyGems Environment:
  - RUBYGEMS VERSION: 3.5.11
  - RUBY VERSION: 3.3.3 (2024-06-12 patchlevel 89) [arm64-darwin23]
  - INSTALLATION DIRECTORY: /Users/lildude/.local/share/gem
  - USER INSTALLATION DIRECTORY: /Users/lildude/.gem/ruby/3.3.0
  - RUBY EXECUTABLE: /Users/lildude/.local/share/mise/installs/ruby/3.3.3/bin/ruby
  - GIT EXECUTABLE: /opt/homebrew/bin/git
  - EXECUTABLE DIRECTORY: /Users/lildude/.local/share/gem/bin
  - SPEC CACHE DIRECTORY: /Users/lildude/.cache/gem
  - SYSTEM CONFIGURATION DIRECTORY: /Users/lildude/.local/share/mise/installs/ruby/3.3.3/etc
  - RUBYGEMS PLATFORMS:
     - ruby
     - arm64-darwin-23
[... truncated for brevity ...]
```

I suspect your env isn't setting the correct `RUBY EXECUTABLE`.

A little bit of Googling led me to https://github.com/conda-forge/ruby-feedstock/issues/116 which indicates you're not alone.

**Reply by sdruskat** [reply]
Thanks! I had had a look but didn't find the issue you dug out.
(And `mamba` is a faster re-implementation and drop-in replacement for `conda`.)
